components/Shipping/basic_usage.rst

```diff
@@ -0,0 +1,657 @@
+
+Setting Shipping Category
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Every shipping method can have shipping category. You can simply set or unset it by calling ``->setCategory()``.
+
+.. code-block:: php
+
+    <?php
+
+        use Sylius\Component\Shipping\Model\ShippingMethod;
+        use Sylius\Component\Shipping\Model\ShippingCategory;
+        use Sylius\Component\Shipping\Model\ShippingMethodInterface;
+
+        $shippingCategory = new ShippingCategory();
+        $shippingCategory->setName('Regular'); // Regular weight items
+
+
+        $shippingMethod = new ShippingMethod();
+        $shippingMethod->setCategory($shippingCategory); //default null, detach
+        $shippingMethod->getCategory(); // Output will be ShippingCategory object
+        $shippingMethod->setCategory(null);
+
+Setting Rule
+~~~~~~~~~~~~
+
+Every shipping method can have many rules, which define its additional requirements. If a **Shipment** does not fulfill
+these requirements (e.g. a rule states that the expected quantity of shipment items should be 2, but the Shipment has
+only one **ShippingItem**), then the **ShippingMethod** having this rule cannot be used on this **Shipment**.
+
+.. code-block:: php
+
+    <?php
+
+        use Sylius\Component\Shipping\Model\Rule;
+        use Sylius\Component\Shipping\Model\ShippingMethod;
+
+        $shippingMethod = new ShippingMethod();
+        $rule = new Rule();
+
+        $shippingMethod->addRule($rule);
+        $shippingMethod->hasRule($rule); // returns true
+        $shippingMethod->getRules(); // collection of rules with count equals 1
+        $shippingMethod->removeRule($rule);
+        $shippingMethod->hasRule($rule); // returns false
+
+Shipping Method Translation
+---------------------------
+
+**ShippingMethodTranslation** allows shipping method's name translation according to given locales. To see how to use translation
+please go to :ref:`component_translation_basic-translations`.
+
+Rule
+----
+
+A **Rule** defines additional requirements for a **ShippingMethod**, which have to be fulfilled by the **Shipment**,
+if it has to be delivered in a way described by this **ShippingMethod**.
+
+.. code-block:: php
+
+    <?php
+
+        use Sylius\Component\Shipping\Model\Rule;
+        use Sylius\Component\Shipping\Model\ShippingMethod;
+        use Sylius\Component\Shipping\Model\RuleInterface;
+
+        $shippingMethod = new ShippingMethod();
+        $rule = new Rule();
+        $rule->setConfiguration(array('count' => 1, 'equal' => true));
+        $rule->setType(RuleInterface::TYPE_ITEM_COUNT);
+        $shippingMethod->addRule($rule);
+
+
+Shipment Item
+-------------
+
+You can use a **ShippingItem** for connecting a shippable object with a proper **Shipment**.
+Note that a **ShippingItem** can exist without a **Shipment** assigned.
+
+.. code-block:: php
+
+    <?php
+
+        use Sylius\Component\Shipping\Model\Shipment;
```

Comment:
Do we really need this additional level of indent everywhere? This is not how you would write this in a php file.